### PR TITLE
Implemented CLI with cobra

### PR DESCRIPTION
**Pull request by soockee**
## Summary
Implemented the CLI program with cobra to be more extendable.
This implementation follows the [kubectl](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/cmd.go) implementation.

## Remarks 
parsing a plan file requries *url* and *filePath*. 
What are those and what are they used for? 

This changes are not tested with actual terraform data.
A exemplary test file or a test case is necessary to do so

### Changelog
* Added cobra for CLI
* Added zerolog for logging
* Added matryer for testing

####  Debug log enabled
```
❯ go run ./cmd/main.go parse -t state -d
{"level":"debug","time":"2023-06-06T18:57:07+02:00","message":"Test"}
{"level":"debug","time":"2023-06-06T18:57:07+02:00","message":"read file "}
```

####   Debug log disabled (default)
```
❯ go run ./cmd/main.go parse -t state 

```

#### Fatal error in command
```
❯ go run ./cmd/main.go parse -t state
{"level":"fatal","error":"type 'state' requires inputPath and outputPath","time":"2023-06-06T19:29:24+02:00"}
exit status 1
```

#### Unsufficient arguments for parse and help text

```
go run ./cmd/main.go parse
Error: required flag(s) "type" not set
Usage:
  karenprocessor parse [flags]

Flags:
      --filePath string   some kind of filepath
  -h, --help              help for parse
  -i, --input string      relative path to input file.
  -o, --output string     relative path to output location.
  -t, --type string       One of 'plan' or 'state'.
      --url string        some kind of url

Global Flags:
  -d, --debug   enables debug log level
```
#### Main program execution prints help text by default

```
❯ go run ./cmd/main.go
karenprocessor - parses terraform state files

Usage:
  karenprocessor [flags]
  karenprocessor [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  parse       parse a terraform state file to karen intermediate format

Flags:
  -d, --debug   enables debug log level
  -h, --help    help for karenprocessor

Use "karenprocessor [command] --help" for more information about a command.
```

### CODE REVIEW

File: pkg/cmd/parse/parse.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			o.Run()
+		},
+	}
+	cmd.Flags().StringVarP(&o.InputType, "type", "t", o.InputType, "One of 'plan' or 'state'.")
+	cmd.Flags().StringVarP(&o.inputPath, "input", "i", o.inputPath, "relative path to input file.")
+	cmd.Flags().StringVarP(&o.outputPath, "output", "o", o.outputPath, "relative path to output location.")
+	// TODO: what kind of URL?
+	cmd.Flags().StringVar(&o.url, "url", o.url, "some kind of url")
+	// TODO: what kind of filepath?
+	cmd.Flags().StringVar(&o.filePath, "filePath", o.filePath, "some kind of filepath")
```
Comment (soockee) [inline pkg/cmd/parse/parse.go]: need a better description

File: pkg/cmd/parse/parse.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+	cmd.MarkFlagRequired("type")
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *Options) Complete(args []string) error {
+	o.args = args
+	log.Debug().Msg("Test")
```
Comment (soockee) [inline pkg/cmd/parse/parse.go]: Remove this log line

File: pkg/cmd/commands.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+	return cmd
+}
+
+func Execute() error {
+	o := NewKarenProcessorOptions()
+	// default command
+	cmd := NewKarenProcessorCommand(o)
+	return cmd.Execute()
+}
```
Comment (soockee) [inline pkg/cmd/commands.go:50]: Add documentation

File: pkg/cmd/commands.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+func runHelp(cmd *cobra.Command, args []string) {
+	cmd.Help()
+}
```
Comment (soockee) [inline pkg/cmd/commands.go:55]: Add documentation

File: pkg/cmd/commands.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		},
+	}
+
+	cmd.PersistentFlags().BoolVarP(&o.debug, "debug", "d", o.debug, "enables debug log level")
+
+	// add sub-commands to root
+	cmd.AddCommand(parse.NewCmdParse())
+
+	return cmd
+}
```
Comment (soockee) [inline pkg/cmd/commands.go:42]: Add documentation

Comment (soockee) [reply pkg/cmd/commands.go:42]: done

File: pkg/cmd/commands.go (hunk 3 above)
Comment (soockee) [reply pkg/cmd/commands.go:50]: done

File: pkg/cmd/commands.go (hunk 4 above)
Comment (soockee) [reply pkg/cmd/commands.go:55]: done

File: pkg/cmd/parse/parse.go (hunk 2 above)
Comment (soockee) [reply pkg/cmd/parse/parse.go]: done

File: pkg/cmd/parse/parse.go (hunk 1 above)
Comment (bfrn) [reply pkg/cmd/parse/parse.go]: I updated the descriptions, if you are fine with them, you can resolve this conversation,

**Review by bfrn — APPROVED** [review]
Nice improvements! 
I updated the descriptions of the `url` and `filePath` flag, if you are okay with them, you can merge.
In the future, these flags can be merged probably into one flag.

**Pull request merged**